components/scream/src/physics/rrtmgp/atmosphere_radiation.cpp

```diff
@@ -850,8 +856,20 @@ void RRTMGPRadiation::run_impl (const int dt) {
     );
 
     // Compute diagnostic total cloud area (vertically-projected cloud cover)
+    auto cldlow = real1d("cldlow", ncol);
+    auto cldmed = real1d("cldmed", ncol);
+    auto cldhgh = real1d("cldhgh", ncol);
     auto cldtot = real1d("cldtot", ncol);
-    rrtmgp::compute_cloud_area(ncol, nlay, nlwgpts, cld_tau_lw_gpt, cldtot);
+    // NOTE: limits for low, mid, and high clouds are mostly taken from EAM F90 source, with the
+    // exception that I removed the restriction on low clouds to be above (numerically lower pressures)
+    // 1200 hPa, and on high clouds to be below (numerically high pressures) 50 hPa. This probably
+    // does not matter in practice, as clouds probably should not be produced above 50 hPa and we
+    // should not be encountering surface pressure above 1200 hPa, but in the event that things go off
+    // the rails we might want to look at these still.
+    rrtmgp::compute_cloud_area(ncol, nlay, nlwgpts, 700e2, std::numeric_limits<Real>::max(), p_lay, cld_tau_lw_gpt, cldlow);
+    rrtmgp::compute_cloud_area(ncol, nlay, nlwgpts, 400e2,                            700e2, p_lay, cld_tau_lw_gpt, cldmed);
+    rrtmgp::compute_cloud_area(ncol, nlay, nlwgpts,     0,                            400e2, p_lay, cld_tau_lw_gpt, cldhgh);
+    rrtmgp::compute_cloud_area(ncol, nlay, nlwgpts,     0, std::numeric_limits<Real>::max(), p_lay, cld_tau_lw_gpt, cldtot);
```

Comment:
Isn't cldtot equal to the sum of the other three? I don't know if this fcn is expensive, but maybe it seems like we have some redundant stuff?

Reply:
Actually, it's not necessarily, that depends on how the subgrid cloud overlaps in the vertical. Imagine this example 3-layer, 3-subcolumn case, where 1s indicate cloud, 0s no cloud, and --- the boundary between low, mid, and high:
```
1 0 0 
-----
1 0 0
-----
1 0 0
```
In this case, each of cldlow, cldmed, and cldhgh would be 1/3, but cldtot would also be 1/3 (project cloud straight down). The only time you'd get cldtot = cldow + cldmed + cldhgh would be the case where each layer had no overlap:
```
1 0 0
-----
0 1 0
-----
0 0 1
```
In which case cldhgh = cldmed = cldow = 1/3, and cldtot = 1.

Reply:
Ok, I don't fully understand the subgrid thingy then. Thanks! =)